### ISSUE TITLE
Create PoC for Client <=> Server Interaction

### ISSUE THREAD

**Issue by pongstylin** [pongstylin/tactics#34]
This involves creating a webpage that has two side-by-side instances of the game loaded that each represent the two players.  The left side is Player 1 and the right side is Player 2.  When it is Player 1's turn, Player 2 is unable to move any units but is capable of selecting/viewing units and visa-versa.  The two instances of the game will communicate with an instance of a virtual server class that will be responsible for echoing actions taken during a turn to the two client instances.'

As part of the refactoring, the offline apps vs bots (Chaos and Faceoff) will work similar to Player vs Player.  That is, the player client instance will communicate with a virtual server instance, which will relay actions between the player and the bot.

All work will be done in the "client" branch until it is complete.  Upon completion, the client branch will be merged into master since it should continue to work even when a true server is absent.

**Comment by pongstylin** [comment]
Added a new commit to the client branch that represents a major refactoring of the client code.  It centers around creating a class to manage game state.  Game state management is something that will be expected of a server, but can also run in the browser for offline games.  In the process of creating this class, I've implemented game data and made some changes to player turn data.  I also made incremental progress on other initiatives such as removing jQuery dependence and converting classes to use ES6 module and class syntax.

Even when running in a browser, however, there needs to be a fairly strong boundary between the client and game state.  Right now, the Game class talks directly to the GameState class.  This is not a terrible thing, but I will introduce a new class called "LocalTransport" that will initialize a Web Worker.  The GameState class will run inside that Web Worker.  The "LocalTransport" class will transparently proxy data exchange between the Game class and the Web Worker.  The benefit of using a Web Worker is to allow the GameState class to run in a separate thread than the Game class.  This will be particularly helpful in allowing the Game class to run smooth animations while bots are thinking hard about their next move in parallel.  Right now, animations will stutter while bots think.  Implementing true parallel thread processing makes the offline game experience operate more similarly to an online game experience.

At some point, we will also create a "RemoteTransport" class that will facilitate data exchange between the Game class and the server.  The Game class does not care if it is talking to a remote or local transport since the interface is expected to be the same.  An instance of either class may be provided to an instance of the Game class so that the Game instance can request and observe changes in state.

**Comment by pongstylin** [comment]
I have changed direction on how to provide a proof-of-concept of client-server communication.  Rather than creating side-by-side apps showing two game clients talking to each other, I have created a web worker that runs the server-side logic in the browser.  The communication between the app and the web worker is similar to the expected communication between the app and a server.

This is now complete and all code has been merged to master.
